src/plugins/data/public/query/filter_manager/filter_manager.ts

```diff
@@ -180,10 +222,15 @@ export class FilterManager {
     this.setFilters([]);
   }
 
-  public static setFiltersStore(filters: esFilters.Filter[], store: esFilters.FilterStateStore) {
+  public static setFiltersStore(
+    filters: esFilters.Filter[],
+    store: esFilters.FilterStateStore,
+    shouldOverrideStore = false
```

Comment:
Why did you have to add this?

Reply:
These make sense to me in case of `setGlobalFilters` and `setAppFilters`. 
This what I was thinking: 
1. In both of those methods we need to handle the case, when filters with incorrect "$store" value are passed in
2. We have couple options then:
 * Just ignore such and do nothing 
 * Throw an error 
 * Override - **I went with this one**.